packages/contractkit/src/wrappers/Exchange.test.ts

```diff
@@ -0,0 +1,55 @@
+import { newKitFromWeb3 } from '../kit'
+import { testWithGanache } from '../test-utils/ganache-test'
+import { ExchangeWrapper } from './Exchange'
+
+/*
+TEST NOTES:
+- In migrations: The only account that has cUSD is accounts[0]
+*/
+
+testWithGanache('Exchange Wrapper', (web3) => {
+  const ONE = web3.utils.toWei('1', 'ether')
+
+  const kit = newKitFromWeb3(web3)
+  let accounts: string[] = []
+  let exchange: ExchangeWrapper
+
+  beforeAll(async () => {
+    accounts = await web3.eth.getAccounts()
+    kit.defaultAccount = accounts[0]
+    exchange = await kit.contracts.getExchange()
+  })
+
+  test('SBAT check buckets', async () => {
+    const [buyBucket, sellBucket] = await exchange.getBuyAndSellBuckets(true)
+    expect(buyBucket).toBeBigNumber()
+    expect(sellBucket).toBeBigNumber()
+    expect(buyBucket.toNumber()).toBeGreaterThan(0)
+    expect(sellBucket.toNumber()).toBeGreaterThan(0)
+  })
+
+  test('SBAT quoteUsdSell', () => expect(exchange.quoteUsdSell(ONE)).resolves.toBeBigNumber())
+  test('SBAT quoteGoldSell', () => expect(exchange.quoteGoldSell(ONE)).resolves.toBeBigNumber())
+  test('SBAT quoteUsdBuy', () => expect(exchange.quoteUsdBuy(ONE)).resolves.toBeBigNumber())
+  test('SBAT quoteGoldBuy', () => expect(exchange.quoteGoldBuy(ONE)).resolves.toBeBigNumber())
+
+  test('SBAT sellDollar', async () => {
+    const goldAmount = await exchange.quoteUsdSell(ONE)
+
+    const stableToken = await kit.contracts.getStableToken()
+    const tx0 = await stableToken.approve(exchange.address, ONE).send()
```

Comment:
Maybe give these meaningful names? `approveTx`, `exchangeTx`.

Reply:
it's not an `approveTx`, is the ERC20's approve, to enable the `transferFrom()` that the Exchange will do.

A different thing would be to add some utility methods to the ExchangeWrapper that "enables" the exchange to move your money. In fact, does our ERC20 implementation have the concept of "inifinte" approval? `approve(MAXINT)`. I remember that's a "thing" in dex

Reply:
I meant renaming `tx0` and `tx` since those names don't carry much meaning.

Reply:
done